compiler/src/dmd/dsymbolsem.d

```diff
@@ -7473,3 +7473,140 @@ private extern(C++) class SetFieldOffsetVisitor : Visitor
         }
     }
 }
+
+// newly imported sem func
+extern(C++) void newScope(Dsymbol d, Scope* sc)
+{
+    scope iav = new newScopeVisitor(sc);
+    d.accept(iav);
+}
+
+extern(C++) class newScopeVisitor : Visitor
+{    
+    //Scope returnScope;
+    
+    alias visit = typeof(super).visit;
+    //alias visit = Visitor.visit;
+    Scope* sc;
+    this(Scope* sc)
+    {
+        this.sc = sc;
+    }
+
+//previously override void visit(Scope* sc)
+override void visit(VisibilityDeclaration atbd)
+    {
+        //Scope returnScope;
+        if (atbd.pkg_identifiers){
+            dsymbolSemantic(atbd, sc);
+        
+            atbd.createNewScope(sc, sc.stc, sc.linkage, sc.cppmangle, atbd.visibility, 1, sc.aligndecl, sc.inlining);
```

Comment:
What you probably want here is 

```
sc = atbd.createNewScope(sc, sc.stc, sc.linkage, sc.cppmangle, atbd.visibility, 1, sc.aligndecl, sc.inlining);
```
and similarly assign to `this.sc` in place of a `return` statement in the original implementation.